datafusion/sql/tests/integration_test.rs

```diff
@@ -527,6 +527,16 @@ fn join_with_ambiguous_column() {
     quick_test(sql, expected);
 }
 
+#[test]
+fn where_selection_with_ambiguous_column() {
+    let sql = "SELECT * FROM person a, person b WHERE id = id + 1";
+    let err = logical_plan(sql).expect_err("query should have failed");
+    assert_eq!(
+        "Internal(\"column reference id is ambiguous\")",
```

Comment:
I think this error should be Plan (not Internal) as it was due to some bad input from the user rather than some bug in datafusion

We can fix that as a follow on PR as well

Reply:
no problem, fixed